design/baremetal-operator/image-builder-integration.md

```diff
@@ -0,0 +1,215 @@
+and use RamDiskImageRequests as required when configured to do so.
+
+## Motivation
+
+When running the ironic-python-agent (IPA) deploy ramdisk, the agent is reliant
+on information baked in to the ramdisk image or available to discover from the
+immediate environment in order to be able to contact Ironic. In most cases this
+involves DHCP supplying an IP address and default route.
+
+Some users would prefer not to run DHCP in their infrastructure, but rather use
+static IP addresses instead. In some cases, particularly when provisioning
+using virtualmedia and not using PXE, they might prefer that the network on
+which to contact Ironic is not an untagged VLAN. In many cases, servers will be
+connected to switches that use static link aggregators, and thus should set up
+NIC bonding to enable reliable two-way communication.
+
+Currently there is no way to use static IPs or configure any of the other
+options on a per-host basis because all hosts are booted using the same ramdisk
+image.
+
+### Goals
+
+- Allow each host to use its own deploy ramdisk image customised with an
+  appropriate network configuration.
+
+### Non-Goals
+
+- Allow users who expect to provision a host to specify their own custom image
+  containing IPA.
+- Use Ironic with images that don't contain IPA.
+- Implement custom provisioners not based on Ironic.
+
+## Proposal
+
+### User Stories
+
+As a data center operator, I would like to manage hosts with metal³ and deploy
+them via virtualmedia on networks where there is no DHCP-like configuration
+service.
+
+## Design Details
+
+A new Custom Resource type, `RamDiskImageRequest` will be created in the
+metal3.io group. The Spec will contain the following fields:
+
+- Name of metaData Secret
+- Name of networkData Secret
+- Architecture
+
+The Status will contain the following fields:
+
+- Image URL
+- Name + Version of metaData Secret
+- Name + Version of networkData Secret
+- Error message
+- Ready condition
+- Error condition
+
+A default controller for this resource type will be implemented that simply
+sets a fixed URL (supplied in its own environment or command line) as the
+output. However, provision will be made for interested parties to implement
+alternate controllers to allow image customisation.
+
+A command-line switch (defaulting to off) will enable integration between the
+baremetal-operator and the image request controller.
+
+### Implementation Details/Notes/Constraints
+
+When integration is enabled, the controller portion of the baremetal-operator
+will update the `RamDiskImageRequest` with the same name and namespace as the
+`BareMetalHost`, creating it if it doesn't exist, to point to the metaData and
+networkData Secrets referenced in the `BareMetalHost` Spec. The Architecture
+will be obtained from the Host's hardware profile.
+
+In the Registration state, the ironic provisioner will avoid attempting to
+Manage the Node in Ironic until the Ready condition is true in the
```

Comment:
I'd prefer we move to `manageable` first. It involves validating the credentials and does not involve any actions with a ramdisk just yet.